tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,162 @@
+"""
+Check the auto-restart feature of containers 
+"""
+import time
+import pytest
+import logging
+
+from common.utilities import wait_until
+from common.helpers.assertions import pytest_assert
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+CMD_CONTAINER_FEATURE_AUTORESTART = "show container feature autorestart"
+
+def get_autorestart_container_list(duthost):
+    """
+    @summary: Get container list by analyzing the command output of 
+              "show container feature autorestart"
+    @return: The list includes containers which was implemented with the autorestart feature
+    """
+    container_list = []
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        if line.split()[1] in ["enabled", "disabled"]:
+            container_list.append(line.split()[0])
+
+    return container_list
+
+
+def get_process_id(duthost, container_name, process):
+    """
+    @summary: Get the pid of process in the specified container by analyzing
+              the command output of "ps -ax"
+    @return: first value is a boolean. True indicates the processs is running
+             second value is the process id. Returning -1 if the process is not found
+    """
+    running_status = False
+    process_id = -1
+    is_running = duthost.shell("docker inspect -f \{\{.State.Running\}\} %s" % container_name)
+    if is_running['stdout'].strip() == "false":
+        return running_status, process_id
```

Comment:
This means the container isn't even running. I feel that this function shouldn't ever be called if the container isn't running. There should be a check prior to calling this function, because there's no point in attempting to test the autorestart feature for a container which isn't running.

Reply:
Great suggestion! Will move this part to the function `kill_process(...)`.

Reply:
moving to kill_process might not be the best place, that means checking this for every process you want to test. You might want to get this state before determine which process(es) to check.

Reply:
@jleveque @yxieca I want to explain why I added this to check whether a container is running or not.  

The first step in function `kill_process(...)` will get the pid of specified process. If the process is running, then it will killed by sending the SIGKILL signal. Before exiting the function `kill_process(...)` to check whether the container is restarted or not, I want to double check whether the process is killed. So at line 68 the `get_process_id(...)` is called again. 

If the process is killed and the container is stopped, then the line 45 in function `get_process_id(...)` will throw an error since the command is sent to a stopped container. So I want to check whether the container is still running before sending the command.

@yxieca gives me a great suggestion. I will create a new function for example `is_container_running(...)` which includes this part and call it in the function `test_autorestart_container(...)`.